addons/hr_recruitment/models/utm_source.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models
+
+from odoo.exceptions import UserError
+
+
+class UtmSource(models.Model):
+    _inherit = 'utm.source'
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_linked_recruitment_sources(self):
```

Comment:
It should already be managed by the "restrict"... is it only for nice error message (to be sure) ? In which case, would maybe add a comment ...